webapp/channels/src/components/profile_popover/profile_localtime.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {DateTime, Duration} from 'luxon';
+
+import {FormattedMessage} from 'react-intl';
+
+import {UserTimezone} from '@mattermost/types/users';
+import Timestamp from 'components/timestamp';
+import {getUserCurrentTimezone} from 'mattermost-redux/utils/timezone_utils';
+
+type ProfileTimezoneProps = {
+    profileUserTimezone?: UserTimezone;
+    currentUserTimezone: string | undefined;
+}
+
+const returnTimeDiff = (
+    currentUserTimezone: string | undefined,
+    profileUserTimezone: UserTimezone | undefined,
+) => {
+    if (!currentUserTimezone || !profileUserTimezone) {
+        return undefined;
+    }
+
+    const currentUserDate = DateTime.local().setZone(currentUserTimezone);
+    let profileUserDate = DateTime.local().setZone('UTC');
+
+    // There is a change that a user is created via API and never logged in, so they do not have any timezone properties set
+    // this defaults to UTC.
+    if (profileUserTimezone?.manualTimezone || profileUserTimezone?.automaticTimezone) {
```

Comment:
Oh, that's interesting. I agree that it seems odd to show a local time if they've never logged in. Can we hide it until they have logged in and their time is set?

The only thing we may have to worry about is that it might mean there is nothing to show in this section of the popover when no local time is available. We will want to avoid showing a blank section or two dividers

Reply:
Yeah, I'm not exactly sure what will show and we can make a separate PR if needed. Since it increases the scope of this slightly. Not sure the usual process. It's a pretty rare occurrence I would think

Reply:
Ya, it's more of an edge case for sure